parsons/etl_helpers/conversions.py

```diff
@@ -0,0 +1,188 @@
+"""
+This module contains functions that can be passed to the Parsons Table convert_column
+method. They take in an original value and return a new value. They can, of course, also be used
+on their own.
+"""
+
+from dateutil.parser import parse
+import datetime
+import re
+
+
+# Datetime conversion functions
+
+
+def date_to_timestamp(value, tzinfo=datetime.timezone.utc):
+    """Convert any date value into a Unix timestamp.
+
+    `Args:`
+        value: int or str or datetime
+            Value to parse
+        tzinfo: datetime.timezone
+            `Optional`: Timezone for the datetime; defaults to UTC.
+    `Returns:`
+        Unix timestamp (int)
+    """
+
+    parsed_date = parse_date(value)
+
+    if not parsed_date:
+        return None
+
+    if not parsed_date.tzinfo:
+        parsed_date = parsed_date.replace(tzinfo=tzinfo)
+
+    return int(parsed_date.timestamp())
+
+
+def parse_date(value, tzinfo=datetime.timezone.utc):
+    """Parse an arbitrary date value into a Python datetime.
+
+    If no value is provided (i.e., the value is None or empty), then the return value will be
+    None.
+
+    `Args:`
+        value: int or str or datetime
+            Value to parse
+        tzinfo: datetime.timezone
+            `Optional`: Timezone for the datetime; defaults to UTC.
+    `Returns:`
+        datetime.datetime or None
+    """
+
+    if not value:
+        return None
+
+    # If it's a number, we (probably) have a unix timestamp
+    if isinstance(value, int):
+        parsed = datetime.datetime.fromtimestamp(value, tzinfo)
+    elif isinstance(value, datetime.datetime):
+        parsed = value
+    elif isinstance(value, str):
+        parsed = parse(value)
+    else:
+        raise TypeError(
+            'Unable to parse value; must be one of string or int or datetime, but got type '
+            f'{type(value)}')
+
+    if not parsed.tzinfo:
+        parsed = parsed.replace(tzinfo=tzinfo)
+
+    return parsed
+
+
+def timestamp_to_readable(value, tzinfo=datetime.timezone.utc, format_as='%Y-%m-%d %H:%M:%S'):
+    """Converts a Unix timestamp into a common human-readable format.
+
+    `Args:`
+        value: int or str or datetime
+            Must correspond to a unix timestamp
+        tzinfo: datetime.timezone
+            `Optional`: Timezone for the datetime; defaults to UTC.
+        format_as: str
+            Should use strftime syntax; defaults to '%Y-%m-%d %H:%M:%S'
+    `Returns:`
+        str
+    """
+    datetime_obj = parse_date(value, tzinfo)
+    return datetime_obj.strftime(format_as)
+
+
+# Flatten contacts
+
+def get_primary_contact_from_nested(contact_list, get_first=True, selector=None):
+    """Extracts single contact value from list of dictionaries.
```

Comment:
I see a shifting in dostring styles between this style (which I prefer) and the description on a new line, e.g.:
```
"""
Extract single contact ...
```

Which is the correct style?

Reply:
I don't think we have a preference. I was going to say "let's go with what Python prefers" but it looks like Python itself [doesn't have a preference](https://peps.python.org/pep-0257/). I know it's a little inconsistent but I don't think it's a big deal to switch back and forth within parsons, and that's one less thing that PR contributors have to worry about.